src/ResourceParser.h

```diff
@@ -475,10 +494,11 @@ namespace snowcrash {
         }
 
         /** Check Parameters eligibility in URI template */
+        template<typename T>
         static void checkParametersEligibility(const MarkdownNodeIterator& node,
                                                const SectionParserData& pd,
                                                Parameters& parameters,
-                                               const ParseResultRef<Resource>& out) {
+                                               const ParseResultRef<T>& out) {
```

Comment:
Is template function good solution there? We can fall in some troubles while someone (in future) will try to specialize it.

Please check following article and be sure about understand of impact http://www.gotw.ca/publications/mill17.htm

Maybe it will be better to switch this into functor implementation via parenthesis operator in structure?